### PR TITLE
Strip irrelevant information from testing docs

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

My plan is to rewrite the testing documentation. However in a first step, this PR removes irrelevant information.

It's simpler to rewrite when I know and we have agreed upon what parts of the current docs are not needed.

I'll comment below as to why I think specific parts are obsolete.

### CODE REVIEW

File: doc/devel/testing.rst [hunk 1]
```diff
@@ -40,11 +47,7 @@ particularly useful ones are:
 ``-v`` or ``--verbose``        Be more verbose
 ``-n NUM``                     Run tests in parallel over NUM
                                processes (requires pytest-xdist_)
-``--timeout=SECONDS``          Set timeout for results from each test
-                               process (requires pytest-timeout_)
```
Comment (timhoffm) [inline doc/devel/testing.rst:44]: Does anybody use `--timeout` explicitly when running the tests? I don't and I don't see why one should.

File: doc/devel/testing.rst [hunk 2]
```diff
@@ -40,11 +47,7 @@ particularly useful ones are:
...
 ``--capture=no`` or ``-s``     Do not capture stdout
-``--flake8``                   Check coding standards using flake8_
-                               (requires pytest-flake8_)
```
Comment (timhoffm) [inline doc/devel/testing.rst:47]: I don't think style should be checked via a `pytest` run (we're also not doing this in CI). The test suite already takes a lot of time.

Will add a note on explicit `flake8` runs in the rewrite.

File: doc/devel/testing.rst [hunk 3]
```diff
@@ -53,15 +56,6 @@ not need to be installed, but Matplotlib should be)::
 
   pytest lib/matplotlib/tests/test_simplification.py::test_clipping
 
-or, if tests are installed, a dot-separated path to the module, optionally
-followed by the function separated by two colons, such as::
-
-  pytest --pyargs matplotlib.tests.test_simplification::test_clipping
```
Comment (timhoffm) [inline doc/devel/testing.rst:59]: There's no benefit in this alternative. It's more to type and you don't have tab-completion for the filename part.

File: doc/devel/testing.rst [hunk 4]
```diff
@@ -53,15 +56,6 @@ not need to be installed, but Matplotlib should be)::
...
-
-If you want to run the full test suite, but want to save wall time try
-running the tests in parallel::
-
-  pytest --verbose -n 5
```
Comment (timhoffm) [inline doc/devel/testing.rst:64]: Information already present via the table of pytest parameters above.

File: doc/devel/testing.rst [hunk 5]
```diff
@@ -106,11 +100,6 @@ For numpy use::
   import numpy as np
   np.random.seed(19680801)
 
-and Python's random number generator::
-
-  import random
-  random.seed(19680801)
-
```
Comment (timhoffm) [inline doc/devel/testing.rst:113]: We don't use builtin `random.seed()` anywhere in our test suite.

File: doc/devel/testing.rst [hunk 6]
```diff
@@ -151,26 +140,6 @@ See the documentation of `~matplotlib.testing.decorators.image_comparison` and
...
-  @pytest.mark.xfail
-  def test_simple_fail():
-      '''very simple example test that should fail'''
-      assert 1 + 1 == 3
-
-Note that the first argument to the :func:`~pytest.mark.xfail` decorator is a
-fail condition, which can be a value such as True, False, or may be a
-dynamically evaluated expression. If a condition is supplied, then a reason
-must also be supplied with the ``reason='message'`` keyword argument.
```
Comment (timhoffm) [inline doc/devel/testing.rst:172]: This is a somewhat lengthy description of a standard pytest feature. Additionally, you will in general not write tests that are expected to fail. So this section is more distracting than helpful.

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
I agree that all these changes are helpful in simplifying.  But its probably major enough that someone like @QuLogic should also take a skim.

File: doc/devel/testing.rst (hunk 3 above)
Comment (QuLogic) [reply doc/devel/testing.rst:59]: It seems losing this part loses the only place that mentions how to run tests on an installed copy.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**